### ISSUE TITLE
[JENKINS-26739] ISE from AbstractLazyLoadRunMap.proposeNewNumber for concurrent matrix builds

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/matrix-project-plugin#606]
<p> Thread has died</p>

<p>java.lang.IllegalStateException: cannot create a build with number 8558 since that (or higher) is already in use among <span class="error">&#91;8099, 8312, 8317, 8318, 8319, 8320, 8321, 8322, 8323, 8326, 8328, 8329, 8330, 8331, 8333, 8335, 8336, 8338, 8340, 8341, 8348, 8351, 8355, 8358, 8360, 8361, 8362, 8363, 8370, 8371, 8380, 8381, 8386, 8387, 8394, 8397, 8398, 8399, 8400, 8401, 8402, 8403, 8404, 8405, 8406, 8407, 8408, 8409, 8418, 8419, 8420, 8421, 8422, 8423, 8424, 8426, 8428, 8435, 8436, 8440, 8441, 8442, 8484, 8487, 8488, 8489, 8490, 8491, 8492, 8493, 8495, 8497, 8498, 8499, 8500, 8501, 8508, 8512, 8513, 8514, 8515, 8522, 8523, 8524, 8526, 8527, 8528, 8529, 8530, 8531, 8535, 8536, 8537, 8545, 8546, 8549, 8550, 8552, 8554, 8555, 8556, 8557, 8560, 8563&#93;</span><br/>
	at jenkins.model.lazy.AbstractLazyLoadRunMap.proposeNewNumber(AbstractLazyLoadRunMap.java:361)<br/>
	at hudson.model.RunMap.put(RunMap.java:189)<br/>
	at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:284)<br/>
	at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:74)<br/>
	at hudson.model.AbstractProject.createExecutable(AbstractProject.java:1205)<br/>
	at hudson.model.AbstractProject.createExecutable(AbstractProject.java:144)<br/>
	at hudson.model.Executor.run(Executor.java:213)</p>

<p>more info</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/deepdiver.png" title="deepdiver's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=deepdiver">deepdiver</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739" target="_blank">ISE from AbstractLazyLoadRunMap.proposeNewNumber for concurrent matrix builds</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick">jglick</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Critical
<li><b>component(s)</b>: matrix-project-plugin
<li><b>label(s)</b>: regression
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2015-08-04T19:02:01+00:00
<li><b>votes</b>: 3
<li><b>watchers</b>: 13
<li><b>imported</b>: 2025-12-06
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p> Thread has died</p>

<p>java.lang.IllegalStateException: cannot create a build with number 8558 since that (or higher) is already in use among <span class="error">&#91;8099, 8312, 8317, 8318, 8319, 8320, 8321, 8322, 8323, 8326, 8328, 8329, 8330, 8331, 8333, 8335, 8336, 8338, 8340, 8341, 8348, 8351, 8355, 8358, 8360, 8361, 8362, 8363, 8370, 8371, 8380, 8381, 8386, 8387, 8394, 8397, 8398, 8399, 8400, 8401, 8402, 8403, 8404, 8405, 8406, 8407, 8408, 8409, 8418, 8419, 8420, 8421, 8422, 8423, 8424, 8426, 8428, 8435, 8436, 8440, 8441, 8442, 8484, 8487, 8488, 8489, 8490, 8491, 8492, 8493, 8495, 8497, 8498, 8499, 8500, 8501, 8508, 8512, 8513, 8514, 8515, 8522, 8523, 8524, 8526, 8527, 8528, 8529, 8530, 8531, 8535, 8536, 8537, 8545, 8546, 8549, 8550, 8552, 8554, 8555, 8556, 8557, 8560, 8563&#93;</span>
	at jenkins.model.lazy.AbstractLazyLoadRunMap.proposeNewNumber(AbstractLazyLoadRunMap.java:361)
	at hudson.model.RunMap.put(RunMap.java:189)
	at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:284)
	at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:74)
	at hudson.model.AbstractProject.createExecutable(AbstractProject.java:1205)
	at hudson.model.AbstractProject.createExecutable(AbstractProject.java:144)
	at hudson.model.Executor.run(Executor.java:213)</p>

<p>more info</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>master on Linux - 2-3 Linux slaves with 10 executors - 1 windows slave with one executor. Problem only shows up on linux slaves where we have the highest load - up to 30 parallel jobs</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
- _Remote link associated with this issue:_


  - [PR 1592](https://github.com/jenkinsci/jenkins/pull/1592)

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=220817&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-220817">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=220817&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-220817">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Possible. Check the file named 'nextBuildNumber' in JENKINS_HOME/jobs/(jobname). It may be as simple as changing its contents.</p>

<p>Anything in the logs to indicate how you got into this situation?</p>
</pre>
</details>

</ul>
</details>

<p>Possible. Check the file named 'nextBuildNumber' in JENKINS_HOME/jobs/(jobname). It may be as simple as changing its contents.</p>

<p>Anything in the logs to indicate how you got into this situation?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/deepdiver.png" title="deepdiver's avatar" /> deepdiver:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=220843&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-220843">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=220843&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-220843">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks for your feedback Daniel.</p>

<p>nextBuildNumber at the moment holds 8610 - which is true having a look at the web console <a href="https://ci.owncloud.org/job/pull-request-analyser-ng-simple/" class="external-link" target="_blank" rel="nofollow noopener">https://ci.owncloud.org/job/pull-request-analyser-ng-simple/</a></p>

<p>To me this feels like some runtime/concurrency issue as soon as too many builds are triggered the same time.
This job is hooked up with github and jobs are being kicked off as soon as a pull request is created or new commits are pushed to the branches.
As desribed in the environment field - this can be up to 30 jobs running in parallel. Under these circumstances the executors die.</p>

<p>Any specific logger category of interest to analyse this issue?</p>

<p>THX</p>
</pre>
</details>

</ul>
</details>

<p>Thanks for your feedback Daniel.</p>

<p>nextBuildNumber at the moment holds 8610 - which is true having a look at the web console <a href="https://ci.owncloud.org/job/pull-request-analyser-ng-simple/" class="external-link" target="_blank" rel="nofollow noopener">https://ci.owncloud.org/job/pull-request-analyser-ng-simple/</a></p>

<p>To me this feels like some runtime/concurrency issue as soon as too many builds are triggered the same time.<br/>
This job is hooked up with github and jobs are being kicked off as soon as a pull request is created or new commits are pushed to the branches.<br/>
As desribed in the environment field - this can be up to 30 jobs running in parallel. Under these circumstances the executors die.</p>

<p>Any specific logger category of interest to analyse this issue?</p>

<p>THX</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=221285&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-221285">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=221285&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-221285">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Builds in question are gone. Would have been interesting to see whether they were created in quick succession (within a few seconds at most).</p>

<p>Any further occurrences of this specific issue?</p>
</pre>
</details>

</ul>
</details>

<p>Builds in question are gone. Would have been interesting to see whether they were created in quick succession (within a few seconds at most).</p>

<p>Any further occurrences of this specific issue?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=221286&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-221286">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=221286&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-221286">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Assigning to <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick" class="user-hover" rel="jglick">jglick</a> as it's related to the new build numbering, may be a concurrency issue when many builds are started in parallel (since they now can be started at a rate of more than 1/second).</p>
</pre>
</details>

</ul>
</details>

<p>Assigning to <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick" class="user-hover" rel="jglick">jglick</a> as it's related to the new build numbering, may be a concurrency issue when many builds are started in parallel (since they now can be started at a rate of more than 1/second).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=221303&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-221303">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=221303&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-221303">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Well <tt>Job.assignBuildNumber</tt> is <tt>synchronized</tt> so new <tt>Run</tt> instances <em>should</em> never collide on number. But some sort of race condition seems like a likely explanation.</p>
</pre>
</details>

</ul>
</details>

<p>Well <tt>Job.assignBuildNumber</tt> is <tt>synchronized</tt> so new <tt>Run</tt> instances <em>should</em> never collide on number. But some sort of race condition seems like a likely explanation.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=221304&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-221304">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=221304&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-221304">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Similar to <a href="https://issue-redirect.jenkins.io/issue/26582" title="ISE from RunMap.put using /git/notifyCommit on a matrix project" class="issue-link" data-issue-key="JENKINS-26582"><del>JENKINS-26582</del></a> but with a different stack trace, which may or may not be significant.</p>

<p>Are you using any plugins which might do funny things with builds—Heavy Job, Gerrit Trigger (known to be buggy unless you install the new beta), etc.?</p>
</pre>
</details>

</ul>
</details>

<p>Similar to <a href="https://issue-redirect.jenkins.io/issue/26582" title="ISE from RunMap.put using /git/notifyCommit on a matrix project" class="issue-link" data-issue-key="JENKINS-26582"><del>JENKINS-26582</del></a> but with a different stack trace, which may or may not be significant.</p>

<p>Are you using any plugins which might do funny things with builds—Heavy Job, Gerrit Trigger (known to be buggy unless you install the new beta), etc.?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/deepdiver.png" title="deepdiver's avatar" /> deepdiver:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=221361&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-221361">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=221361&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-221361">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The job in question is based on the github pull request builder - might that be the reason?</p>

<p>Could a quite period help? Let me try that ...</p>
</pre>
</details>

</ul>
</details>

<p>The job in question is based on the github pull request builder - might that be the reason?</p>

<p>Could a quite period help? Let me try that ...</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/deepdiver.png" title="deepdiver's avatar" /> deepdiver:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=221478&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-221478">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=221478&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-221478">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>okay - quite period doesn't help ... feel like a bit less - but we are not that acive at the moment.</p>

<p>Any further ideas on this? THX</p>
</pre>
</details>

</ul>
</details>

<p>okay - quite period doesn't help ... feel like a bit less - but we are not that acive at the moment.</p>

<p>Any further ideas on this? THX</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dmaslakov.png" title="dmaslakov's avatar" /> dmaslakov:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=221995&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-221995">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=221995&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-221995">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>After upgrade from 1.596 to 1.599 I got same issues. It happened on matrix job.</p>

<p>Here is a log, see how same build numbers are probed several times and failed:</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>Feb 18, 2015 6:55:09 AM hudson.model.Executor run
SEVERE: Unexpected executor death
java.lang.IllegalStateException: cannot create a build with number 2439 since that (or higher) is already in use among [2376, 2377, 2378, 2379, 2380, 2381, 2382, 2383, 2384, 2385, 2386, 2387, 2388, 2389, 2390, 2391, 2392, 2393, 2394, 2395, 2396, 2397, 2398, 2399, 2400, 2401, 2402, 2403, 2404, 2405, 2406, 2407, 2408, 2409, 2410, 2411, 2412, 2413, 2414, 2415, 2416, 2417, 2418, 2419, 2420, 2421, 2422, 2423, 2424, 2425, 2426, 2427, 2428, 2429, 2430, 2431, 2432, 2433, 2434, 2435, 2436, 2441]
        at jenkins.model.lazy.AbstractLazyLoadRunMap.proposeNewNumber(AbstractLazyLoadRunMap.java:361)
        at hudson.model.RunMap.put(RunMap.java:189)
        at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:284)
        at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:74)
        at hudson.model.AbstractProject.createExecutable(AbstractProject.java:1205)
        at hudson.model.AbstractProject.createExecutable(AbstractProject.java:144)
        at hudson.model.Executor.run(Executor.java:213)

Feb 18, 2015 6:55:09 AM hudson.model.Executor run
SEVERE: Unexpected executor death
java.lang.IllegalStateException: cannot create a build with number 2440 since that (or higher) is already in use among [2376, 2377, 2378, 2379, 2380, 2381, 2382, 2383, 2384, 2385, 2386, 2387, 2388, 2389, 2390, 2391, 2392, 2393, 2394, 2395, 2396, 2397, 2398, 2399, 2400, 2401, 2402, 2403, 2404, 2405, 2406, 2407, 2408, 2409, 2410, 2411, 2412, 2413, 2414, 2415, 2416, 2417, 2418, 2419, 2420, 2421, 2422, 2423, 2424, 2425, 2426, 2427, 2428, 2429, 2430, 2431, 2432, 2433, 2434, 2435, 2436, 2441]
        at jenkins.model.lazy.AbstractLazyLoadRunMap.proposeNewNumber(AbstractLazyLoadRunMap.java:361)
        at hudson.model.RunMap.put(RunMap.java:189)
        at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:284)
        at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:74)
        at hudson.model.AbstractProject.createExecutable(AbstractProject.java:1205)
        at hudson.model.AbstractProject.createExecutable(AbstractProject.java:144)
        at hudson.model.Executor.run(Executor.java:213)

Feb 18, 2015 6:55:14 AM hudson.model.Executor run
SEVERE: Unexpected executor death
java.lang.IllegalStateException: cannot create a build with number 2439 since that (or higher) is already in use among [2379, 2380, 2381, 2382, 2383, 2384, 2385, 2386, 2387, 2388, 2389, 2390, 2391, 2392, 2393, 2394, 2395, 2396, 2397, 2398, 2399, 2400, 2401, 2402, 2403, 2404, 2405, 2406, 2407, 2408, 2409, 2410, 2411, 2412, 2413, 2414, 2415, 2416, 2417, 2418, 2419, 2420, 2421, 2422, 2423, 2424, 2425, 2426, 2427, 2428, 2429, 2430, 2431, 2432, 2433, 2434, 2435, 2436, 2437, 2438, 2441]
        at jenkins.model.lazy.AbstractLazyLoadRunMap.proposeNewNumber(AbstractLazyLoadRunMap.java:361)
        at hudson.model.RunMap.put(RunMap.java:189)
        at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:284)
        at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:74)
        at hudson.model.AbstractProject.createExecutable(AbstractProject.java:1205)
        at hudson.model.AbstractProject.createExecutable(AbstractProject.java:144)
        at hudson.model.Executor.run(Executor.java:213)

Feb 18, 2015 6:55:15 AM hudson.model.Executor run
SEVERE: Unexpected executor death
java.lang.IllegalStateException: cannot create a build with number 2440 since that (or higher) is already in use among [2379, 2380, 2381, 2382, 2383, 2384, 2385, 2386, 2387, 2388, 2389, 2390, 2391, 2392, 2393, 2394, 2395, 2396, 2397, 2398, 2399, 2400, 2401, 2402, 2403, 2404, 2405, 2406, 2407, 2408, 2409, 2410, 2411, 2412, 2413, 2414, 2415, 2416, 2417, 2418, 2419, 2420, 2421, 2422, 2423, 2424, 2425, 2426, 2427, 2428, 2429, 2430, 2431, 2432, 2433, 2434, 2435, 2436, 2437, 2438, 2441]
        at jenkins.model.lazy.AbstractLazyLoadRunMap.proposeNewNumber(AbstractLazyLoadRunMap.java:361)
        at hudson.model.RunMap.put(RunMap.java:189)
        at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:284)
        at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:74)
        at hudson.model.AbstractProject.createExecutable(AbstractProject.java:1205)
        at hudson.model.AbstractProject.createExecutable(AbstractProject.java:144)
        at hudson.model.Executor.run(Executor.java:213)

Feb 18, 2015 6:55:19 AM hudson.model.Executor run
SEVERE: Unexpected executor death
java.lang.IllegalStateException: cannot create a build with number 2439 since that (or higher) is already in use among [2376, 2377, 2378, 2379, 2380, 2381, 2382, 2383, 2384, 2385, 2386, 2387, 2388, 2389, 2390, 2391, 2392, 2393, 2394, 2395, 2396, 2397, 2398, 2399, 2400, 2401, 2402, 2403, 2404, 2405, 2406, 2407, 2408, 2409, 2410, 2411, 2412, 2413, 2414, 2415, 2416, 2417, 2418, 2419, 2420, 2421, 2422, 2423, 2424, 2425, 2426, 2427, 2428, 2429, 2430, 2431, 2432, 2433, 2434, 2435, 2436, 2437, 2438, 2441]
        at jenkins.model.lazy.AbstractLazyLoadRunMap.proposeNewNumber(AbstractLazyLoadRunMap.java:361)
        at hudson.model.RunMap.put(RunMap.java:189)
        at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:284)
        at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:74)
        at hudson.model.AbstractProject.createExecutable(AbstractProject.java:1205)
        at hudson.model.AbstractProject.createExecutable(AbstractProject.java:144)
        at hudson.model.Executor.run(Executor.java:213)

Feb 18, 2015 6:55:20 AM hudson.model.Executor run
SEVERE: Unexpected executor death
java.lang.IllegalStateException: cannot create a build with number 2440 since that (or higher) is already in use among [2376, 2377, 2378, 2379, 2380, 2381, 2382, 2383, 2384, 2385, 2386, 2387, 2388, 2389, 2390, 2391, 2392, 2393, 2394, 2395, 2396, 2397, 2398, 2399, 2400, 2401, 2402, 2403, 2404, 2405, 2406, 2407, 2408, 2409, 2410, 2411, 2412, 2413, 2414, 2415, 2416, 2417, 2418, 2419, 2420, 2421, 2422, 2423, 2424, 2425, 2426, 2427, 2428, 2429, 2430, 2431, 2432, 2433, 2434, 2435, 2436, 2437, 2438, 2441]
        at jenkins.model.lazy.AbstractLazyLoadRunMap.proposeNewNumber(AbstractLazyLoadRunMap.java:361)
        at hudson.model.RunMap.put(RunMap.java:189)
        at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:284)
        at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:74)
        at hudson.model.AbstractProject.createExecutable(AbstractProject.java:1205)
        at hudson.model.AbstractProject.createExecutable(AbstractProject.java:144)
        at hudson.model.Executor.run(Executor.java:213)
</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>After upgrade from 1.596 to 1.599 I got same issues. It happened on matrix job.</p>

<p>Here is a log, see how same build numbers are probed several times and failed:</p>



```
Feb 18, 2015 6:55:09 AM hudson.model.Executor run
SEVERE: Unexpected executor death
java.lang.IllegalStateException: cannot create a build with number 2439 since that (or higher) is already in use among [2376, 2377, 2378, 2379, 2380, 2381, 2382, 2383, 2384, 2385, 2386, 2387, 2388, 2389, 2390, 2391, 2392, 2393, 2394, 2395, 2396, 2397, 2398, 2399, 2400, 2401, 2402, 2403, 2404, 2405, 2406, 2407, 2408, 2409, 2410, 2411, 2412, 2413, 2414, 2415, 2416, 2417, 2418, 2419, 2420, 2421, 2422, 2423, 2424, 2425, 2426, 2427, 2428, 2429, 2430, 2431, 2432, 2433, 2434, 2435, 2436, 2441]
at jenkins.model.lazy.AbstractLazyLoadRunMap.proposeNewNumber(AbstractLazyLoadRunMap.java:361)
at hudson.model.RunMap.put(RunMap.java:189)
at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:284)
at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:74)
at hudson.model.AbstractProject.createExecutable(AbstractProject.java:1205)
at hudson.model.AbstractProject.createExecutable(AbstractProject.java:144)
at hudson.model.Executor.run(Executor.java:213)

Feb 18, 2015 6:55:09 AM hudson.model.Executor run
SEVERE: Unexpected executor death
java.lang.IllegalStateException: cannot create a build with number 2440 since that (or higher) is already in use among [2376, 2377, 2378, 2379, 2380, 2381, 2382, 2383, 2384, 2385, 2386, 2387, 2388, 2389, 2390, 2391, 2392, 2393, 2394, 2395, 2396, 2397, 2398, 2399, 2400, 2401, 2402, 2403, 2404, 2405, 2406, 2407, 2408, 2409, 2410, 2411, 2412, 2413, 2414, 2415, 2416, 2417, 2418, 2419, 2420, 2421, 2422, 2423, 2424, 2425, 2426, 2427, 2428, 2429, 2430, 2431, 2432, 2433, 2434, 2435, 2436, 2441]
at jenkins.model.lazy.AbstractLazyLoadRunMap.proposeNewNumber(AbstractLazyLoadRunMap.java:361)
at hudson.model.RunMap.put(RunMap.java:189)
at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:284)
at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:74)
at hudson.model.AbstractProject.createExecutable(AbstractProject.java:1205)
at hudson.model.AbstractProject.createExecutable(AbstractProject.java:144)
at hudson.model.Executor.run(Executor.java:213)

Feb 18, 2015 6:55:14 AM hudson.model.Executor run
SEVERE: Unexpected executor death
java.lang.IllegalStateException: cannot create a build with number 2439 since that (or higher) is already in use among [2379, 2380, 2381, 2382, 2383, 2384, 2385, 2386, 2387, 2388, 2389, 2390, 2391, 2392, 2393, 2394, 2395, 2396, 2397, 2398, 2399, 2400, 2401, 2402, 2403, 2404, 2405, 2406, 2407, 2408, 2409, 2410, 2411, 2412, 2413, 2414, 2415, 2416, 2417, 2418, 2419, 2420, 2421, 2422, 2423, 2424, 2425, 2426, 2427, 2428, 2429, 2430, 2431, 2432, 2433, 2434, 2435, 2436, 2437, 2438, 2441]
at jenkins.model.lazy.AbstractLazyLoadRunMap.proposeNewNumber(AbstractLazyLoadRunMap.java:361)
at hudson.model.RunMap.put(RunMap.java:189)
at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:284)
at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:74)
at hudson.model.AbstractProject.createExecutable(AbstractProject.java:1205)
at hudson.model.AbstractProject.createExecutable(AbstractProject.java:144)
at hudson.model.Executor.run(Executor.java:213)

Feb 18, 2015 6:55:15 AM hudson.model.Executor run
SEVERE: Unexpected executor death
java.lang.IllegalStateException: cannot create a build with number 2440 since that (or higher) is already in use among [2379, 2380, 2381, 2382, 2383, 2384, 2385, 2386, 2387, 2388, 2389, 2390, 2391, 2392, 2393, 2394, 2395, 2396, 2397, 2398, 2399, 2400, 2401, 2402, 2403, 2404, 2405, 2406, 2407, 2408, 2409, 2410, 2411, 2412, 2413, 2414, 2415, 2416, 2417, 2418, 2419, 2420, 2421, 2422, 2423, 2424, 2425, 2426, 2427, 2428, 2429, 2430, 2431, 2432, 2433, 2434, 2435, 2436, 2437, 2438, 2441]
at jenkins.model.lazy.AbstractLazyLoadRunMap.proposeNewNumber(AbstractLazyLoadRunMap.java:361)
at hudson.model.RunMap.put(RunMap.java:189)
at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:284)
at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:74)
at hudson.model.AbstractProject.createExecutable(AbstractProject.java:1205)
at hudson.model.AbstractProject.createExecutable(AbstractProject.java:144)
at hudson.model.Executor.run(Executor.java:213)

Feb 18, 2015 6:55:19 AM hudson.model.Executor run
SEVERE: Unexpected executor death
java.lang.IllegalStateException: cannot create a build with number 2439 since that (or higher) is already in use among [2376, 2377, 2378, 2379, 2380, 2381, 2382, 2383, 2384, 2385, 2386, 2387, 2388, 2389, 2390, 2391, 2392, 2393, 2394, 2395, 2396, 2397, 2398, 2399, 2400, 2401, 2402, 2403, 2404, 2405, 2406, 2407, 2408, 2409, 2410, 2411, 2412, 2413, 2414, 2415, 2416, 2417, 2418, 2419, 2420, 2421, 2422, 2423, 2424, 2425, 2426, 2427, 2428, 2429, 2430, 2431, 2432, 2433, 2434, 2435, 2436, 2437, 2438, 2441]
at jenkins.model.lazy.AbstractLazyLoadRunMap.proposeNewNumber(AbstractLazyLoadRunMap.java:361)
at hudson.model.RunMap.put(RunMap.java:189)
at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:284)
at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:74)
at hudson.model.AbstractProject.createExecutable(AbstractProject.java:1205)
at hudson.model.AbstractProject.createExecutable(AbstractProject.java:144)
at hudson.model.Executor.run(Executor.java:213)

Feb 18, 2015 6:55:20 AM hudson.model.Executor run
SEVERE: Unexpected executor death
java.lang.IllegalStateException: cannot create a build with number 2440 since that (or higher) is already in use among [2376, 2377, 2378, 2379, 2380, 2381, 2382, 2383, 2384, 2385, 2386, 2387, 2388, 2389, 2390, 2391, 2392, 2393, 2394, 2395, 2396, 2397, 2398, 2399, 2400, 2401, 2402, 2403, 2404, 2405, 2406, 2407, 2408, 2409, 2410, 2411, 2412, 2413, 2414, 2415, 2416, 2417, 2418, 2419, 2420, 2421, 2422, 2423, 2424, 2425, 2426, 2427, 2428, 2429, 2430, 2431, 2432, 2433, 2434, 2435, 2436, 2437, 2438, 2441]
at jenkins.model.lazy.AbstractLazyLoadRunMap.proposeNewNumber(AbstractLazyLoadRunMap.java:361)
at hudson.model.RunMap.put(RunMap.java:189)
at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:284)
at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:74)
at hudson.model.AbstractProject.createExecutable(AbstractProject.java:1205)
at hudson.model.AbstractProject.createExecutable(AbstractProject.java:144)
at hudson.model.Executor.run(Executor.java:213)

```

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>eugene:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=222040&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222040">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=222040&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222040">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi All,
For the issue resolving I downgraded Jenkins version to 1.595.
This proves that the problem in the Jenkins core.</p>
</pre>
</details>

</ul>
</details>

<p>Hi All,<br/>
For the issue resolving I downgraded Jenkins version to 1.595.<br/>
This proves that the problem in the Jenkins core.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=222065&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222065">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=222065&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222065">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The bug appears to be in Matrix Plugin (where it creates MatrixRuns for MatrixConfigurations with build numbers synchronized with their parent MatrixBuild) but the new, stricter validation in Jenkins core results in the broken behavior now having actual consequences.</p>
</pre>
</details>

</ul>
</details>

<p>The bug appears to be in Matrix Plugin (where it creates MatrixRuns for MatrixConfigurations with build numbers synchronized with their parent MatrixBuild) but the new, stricter validation in Jenkins core results in the broken behavior now having actual consequences.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=222127&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222127">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=222127&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222127">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I see, the overloads of <tt>getNextBuildNumber</tt> and <tt>assignBuildNumber</tt> in <tt>MatrixConfiguration</tt> are rather dangerous.</p>
</pre>
</details>

</ul>
</details>

<p>I see, the overloads of <tt>getNextBuildNumber</tt> and <tt>assignBuildNumber</tt> in <tt>MatrixConfiguration</tt> are rather dangerous.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=222796&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222796">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=222796&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222796">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The good news is that after updating the plugin’s baseline to 1.600, a similar error appears for example in <tt>MatrixProjectTest.testConcurrentBuild</tt>, which means I have something concrete to go on.</p>
</pre>
</details>

</ul>
</details>

<p>The good news is that after updating the plugin’s baseline to 1.600, a similar error appears for example in <tt>MatrixProjectTest.testConcurrentBuild</tt>, which means I have something concrete to go on.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=222850&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222850">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=222850&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222850">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Jesse Glick
Path:
 src/test/java/hudson/matrix/MatrixProjectTest.java
<a href="http://jenkins-ci.org/commit/matrix-project-plugin/5f011e8f2718937295584cdf1101941631d6e7ec" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/matrix-project-plugin/5f011e8f2718937295584cdf1101941631d6e7ec</a>
Log:
  <a href="https://issue-redirect.jenkins.io/issue/26739" title="ISE from AbstractLazyLoadRunMap.proposeNewNumber for concurrent matrix builds" class="issue-link" data-issue-key="JENKINS-26739"><del>JENKINS-26739</del></a> Suppress testConcurrentBuild until we can pick up a fix.</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Jesse Glick<br/>
Path:<br/>
 src/test/java/hudson/matrix/MatrixProjectTest.java<br/>
<a href="http://jenkins-ci.org/commit/matrix-project-plugin/5f011e8f2718937295584cdf1101941631d6e7ec" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/matrix-project-plugin/5f011e8f2718937295584cdf1101941631d6e7ec</a><br/>
Log:<br/>
  <a href="https://issue-redirect.jenkins.io/issue/26739" title="ISE from AbstractLazyLoadRunMap.proposeNewNumber for concurrent matrix builds" class="issue-link" data-issue-key="JENKINS-26739"><del>JENKINS-26739</del></a> Suppress testConcurrentBuild until we can pick up a fix.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=222858&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222858">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=222858&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222858">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Jesse Glick
Path:
 core/src/main/java/hudson/model/RunMap.java
<a href="http://jenkins-ci.org/commit/jenkins/5ada903554a493fcfbc8e23c5c8d02bc50c17845" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/5ada903554a493fcfbc8e23c5c8d02bc50c17845</a>
Log:
  <span class="error">&#91;FIXED JENKINS-26739&#93;</span> Suppress monotonicity assertion for MatrixRun.</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Jesse Glick<br/>
Path:<br/>
 core/src/main/java/hudson/model/RunMap.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/5ada903554a493fcfbc8e23c5c8d02bc50c17845" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/5ada903554a493fcfbc8e23c5c8d02bc50c17845</a><br/>
Log:<br/>
  <span class="error">&#91;FIXED JENKINS-26739&#93;</span> Suppress monotonicity assertion for MatrixRun.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=222859&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222859">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=222859&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222859">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Jesse Glick
Path:
 changelog.html
 core/src/main/java/hudson/model/RunMap.java
<a href="http://jenkins-ci.org/commit/jenkins/cf2b02dc58c140aafc4e20115a8bc936d05815be" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/cf2b02dc58c140aafc4e20115a8bc936d05815be</a>
Log:
  <a href="https://issue-redirect.jenkins.io/issue/26739" title="ISE from AbstractLazyLoadRunMap.proposeNewNumber for concurrent matrix builds" class="issue-link" data-issue-key="JENKINS-26739"><del>JENKINS-26739</del></a> Merging #1592.</p>


<p>Compare: <a href="https://github.com/jenkinsci/jenkins/compare/87ab95e7fc30...cf2b02dc58c1" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/87ab95e7fc30...cf2b02dc58c1</a></p>

</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Jesse Glick<br/>
Path:<br/>
 changelog.html<br/>
 core/src/main/java/hudson/model/RunMap.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/cf2b02dc58c140aafc4e20115a8bc936d05815be" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/cf2b02dc58c140aafc4e20115a8bc936d05815be</a><br/>
Log:<br/>
  <a href="https://issue-redirect.jenkins.io/issue/26739" title="ISE from AbstractLazyLoadRunMap.proposeNewNumber for concurrent matrix builds" class="issue-link" data-issue-key="JENKINS-26739"><del>JENKINS-26739</del></a> Merging #1592.</p>


<p>Compare: <a href="https://github.com/jenkinsci/jenkins/compare/87ab95e7fc30...cf2b02dc58c1" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/87ab95e7fc30...cf2b02dc58c1</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=222860&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222860">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=222860&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222860">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Jesse Glick
Path:
 changelog.html
 core/src/main/java/hudson/model/RunMap.java
<a href="http://jenkins-ci.org/commit/jenkins/6ee6fc3b2f7a9bceca1b04c4af5e6f1b024e2ab4" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/6ee6fc3b2f7a9bceca1b04c4af5e6f1b024e2ab4</a>
Log:
  <a href="https://issue-redirect.jenkins.io/issue/26739" title="ISE from AbstractLazyLoadRunMap.proposeNewNumber for concurrent matrix builds" class="issue-link" data-issue-key="JENKINS-26739"><del>JENKINS-26739</del></a> Backporting #1592 to rc.</p>

<p>(cherry picked from commit cf2b02dc58c140aafc4e20115a8bc936d05815be)</p>

<p>Conflicts:
	changelog.html</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Jesse Glick<br/>
Path:<br/>
 changelog.html<br/>
 core/src/main/java/hudson/model/RunMap.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/6ee6fc3b2f7a9bceca1b04c4af5e6f1b024e2ab4" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/6ee6fc3b2f7a9bceca1b04c4af5e6f1b024e2ab4</a><br/>
Log:<br/>
  <a href="https://issue-redirect.jenkins.io/issue/26739" title="ISE from AbstractLazyLoadRunMap.proposeNewNumber for concurrent matrix builds" class="issue-link" data-issue-key="JENKINS-26739"><del>JENKINS-26739</del></a> Backporting #1592 to rc.</p>

<p>(cherry picked from commit cf2b02dc58c140aafc4e20115a8bc936d05815be)</p>

<p>Conflicts:<br/>
	changelog.html</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=222861&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222861">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=222861&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222861">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Should be in 1.601.</p>
</pre>
</details>

</ul>
</details>

<p>Should be in 1.601.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dogfood.png" title="dogfood's avatar" /> dogfood:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=222862&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222862">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=222862&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222862">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/blue.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/3995/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #3995</a>
     <span class="error">&#91;FIXED JENKINS-26739&#93;</span> Suppress monotonicity assertion for MatrixRun. (Revision 5ada903554a493fcfbc8e23c5c8d02bc50c17845)</p>

<p>     Result = SUCCESS
jesse glick : <a href="https://github.com/jenkinsci/jenkins/commit/5ada903554a493fcfbc8e23c5c8d02bc50c17845" class="external-link" target="_blank" rel="nofollow noopener">5ada903554a493fcfbc8e23c5c8d02bc50c17845</a>
Files : </p>
<ul>
	<li>core/src/main/java/hudson/model/RunMap.java</li>
</ul>

</pre>
</details>

</ul>
</details>

<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/blue.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/3995/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #3995</a><br/>
     <span class="error">&#91;FIXED JENKINS-26739&#93;</span> Suppress monotonicity assertion for MatrixRun. (Revision 5ada903554a493fcfbc8e23c5c8d02bc50c17845)</p>

<p>     Result = SUCCESS<br/>
jesse glick : <a href="https://github.com/jenkinsci/jenkins/commit/5ada903554a493fcfbc8e23c5c8d02bc50c17845" class="external-link" target="_blank" rel="nofollow noopener">5ada903554a493fcfbc8e23c5c8d02bc50c17845</a><br/>
Files : </p>
<ul>
	<li>core/src/main/java/hudson/model/RunMap.java</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dogfood.png" title="dogfood's avatar" /> dogfood:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=222917&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222917">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=222917&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-222917">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/blue.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/3997/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #3997</a>
     <a href="https://issue-redirect.jenkins.io/issue/26739" title="ISE from AbstractLazyLoadRunMap.proposeNewNumber for concurrent matrix builds" class="issue-link" data-issue-key="JENKINS-26739"><del>JENKINS-26739</del></a> Backporting #1592 to rc. (Revision 6ee6fc3b2f7a9bceca1b04c4af5e6f1b024e2ab4)</p>

<p>     Result = SUCCESS
jesse glick : <a href="https://github.com/jenkinsci/jenkins/commit/6ee6fc3b2f7a9bceca1b04c4af5e6f1b024e2ab4" class="external-link" target="_blank" rel="nofollow noopener">6ee6fc3b2f7a9bceca1b04c4af5e6f1b024e2ab4</a>
Files : </p>
<ul>
	<li>changelog.html</li>
	<li>core/src/main/java/hudson/model/RunMap.java</li>
</ul>

</pre>
</details>

</ul>
</details>

<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/blue.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/3997/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #3997</a><br/>
     <a href="https://issue-redirect.jenkins.io/issue/26739" title="ISE from AbstractLazyLoadRunMap.proposeNewNumber for concurrent matrix builds" class="issue-link" data-issue-key="JENKINS-26739"><del>JENKINS-26739</del></a> Backporting #1592 to rc. (Revision 6ee6fc3b2f7a9bceca1b04c4af5e6f1b024e2ab4)</p>

<p>     Result = SUCCESS<br/>
jesse glick : <a href="https://github.com/jenkinsci/jenkins/commit/6ee6fc3b2f7a9bceca1b04c4af5e6f1b024e2ab4" class="external-link" target="_blank" rel="nofollow noopener">6ee6fc3b2f7a9bceca1b04c4af5e6f1b024e2ab4</a><br/>
Files : </p>
<ul>
	<li>changelog.html</li>
	<li>core/src/main/java/hudson/model/RunMap.java</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/deepdiver.png" title="deepdiver's avatar" /> deepdiver:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=223384&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-223384">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=223384&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-223384">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I have not seen this issue anymore since we upgraded to 1.600 - many thanks!</p>
</pre>
</details>

</ul>
</details>

<p>I have not seen this issue anymore since we upgraded to 1.600 - many thanks!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>gbougeard:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=224364&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-224364">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=224364&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-224364">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi, we currently have this issue with 1.605 and 1.606 version.</p>
</pre>
</details>

</ul>
</details>

<p>Hi, we currently have this issue with 1.605 and 1.606 version.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=224371&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-224371">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=224371&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-224371">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=gbougeard" class="user-hover" rel="gbougeard">gbougeard</a> really with the same stack trace? Seems more likely that you are seeing some other issue with a similar but distinct symptom. File it separately, with steps to reproduce if at all possible (otherwise at least a log file, support bundle from the Support Core plugin, etc.), blocking <a href="https://issue-redirect.jenkins.io/issue/24380" title="Use build numbers as IDs" class="issue-link" data-issue-key="JENKINS-24380"><del>JENKINS-24380</del></a>.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=gbougeard" class="user-hover" rel="gbougeard">gbougeard</a> really with the same stack trace? Seems more likely that you are seeing some other issue with a similar but distinct symptom. File it separately, with steps to reproduce if at all possible (otherwise at least a log file, support bundle from the Support Core plugin, etc.), blocking <a href="https://issue-redirect.jenkins.io/issue/24380" title="Use build numbers as IDs" class="issue-link" data-issue-key="JENKINS-24380"><del>JENKINS-24380</del></a>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>gbougeard:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=224372&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-224372">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=224372&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-224372">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Do you think it's a different issue?</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">  Mar 25, 2015 12:57:18 PM SEVERE hudson.model.Executor run
  Unexpected executor death
  java.lang.IllegalStateException: /<span class="code-keyword">var</span>/lib/jenkins/jobs/service-mysql-migrations_master/configurations/axis-BASE_TAG/prod/builds/218 already existed; will not overwite with service-mysql-migrations_master/BASE_TAG=prod #218
	at hudson.model.RunMap.put(RunMap.java:187)
	at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:284)
	at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:74)
	at hudson.model.AbstractProject.createExecutable(AbstractProject.java:1205)
	at hudson.model.AbstractProject.createExecutable(AbstractProject.java:144)
</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>Do you think it's a different issue?</p>

<pre>
  Mar 25, 2015 12:57:18 PM SEVERE hudson.model.Executor run
  Unexpected executor death
  java.lang.IllegalStateException: /<span class="code-keyword">var</span>/lib/jenkins/jobs/service-mysql-migrations_master/configurations/axis-BASE_TAG/prod/builds/218 already existed; will not overwite with service-mysql-migrations_master/BASE_TAG=prod #218
	at hudson.model.RunMap.put(RunMap.java:187)
	at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:284)
	at hudson.matrix.MatrixConfiguration.newBuild(MatrixConfiguration.java:74)
	at hudson.model.AbstractProject.createExecutable(AbstractProject.java:1205)
	at hudson.model.AbstractProject.createExecutable(AbstractProject.java:144)
</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=224375&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-224375">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=224375&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-224375">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=gbougeard" class="user-hover" rel="gbougeard">gbougeard</a> Correct, that is unrelated, and already tracked as <a href="https://issue-redirect.jenkins.io/issue/26582" title="ISE from RunMap.put using /git/notifyCommit on a matrix project" class="issue-link" data-issue-key="JENKINS-26582"><del>JENKINS-26582</del></a> (currently with no known way to reproduce).</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=gbougeard" class="user-hover" rel="gbougeard">gbougeard</a> Correct, that is unrelated, and already tracked as <a href="https://issue-redirect.jenkins.io/issue/26582" title="ISE from RunMap.put using /git/notifyCommit on a matrix project" class="issue-link" data-issue-key="JENKINS-26582"><del>JENKINS-26582</del></a> (currently with no known way to reproduce).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=229909&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-229909">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=229909&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-229909">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Jesse Glick
Path:
 src/test/java/hudson/matrix/MatrixProjectTest.java
<a href="http://jenkins-ci.org/commit/matrix-project-plugin/66592112c2984ecd379cf6ff0ab8e1eed68dfa60" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/matrix-project-plugin/66592112c2984ecd379cf6ff0ab8e1eed68dfa60</a>
Log:
  <a href="https://issue-redirect.jenkins.io/issue/26739" title="ISE from AbstractLazyLoadRunMap.proposeNewNumber for concurrent matrix builds" class="issue-link" data-issue-key="JENKINS-26739"><del>JENKINS-26739</del></a> testConcurrentBuild should pass again as of 1.602.</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Jesse Glick<br/>
Path:<br/>
 src/test/java/hudson/matrix/MatrixProjectTest.java<br/>
<a href="http://jenkins-ci.org/commit/matrix-project-plugin/66592112c2984ecd379cf6ff0ab8e1eed68dfa60" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/matrix-project-plugin/66592112c2984ecd379cf6ff0ab8e1eed68dfa60</a><br/>
Log:<br/>
  <a href="https://issue-redirect.jenkins.io/issue/26739" title="ISE from AbstractLazyLoadRunMap.proposeNewNumber for concurrent matrix builds" class="issue-link" data-issue-key="JENKINS-26739"><del>JENKINS-26739</del></a> testConcurrentBuild should pass again as of 1.602.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>dsajner:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=233758&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-233758">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=233758&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-233758">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Reopening this issue.  I'm seeing this exception and a lot of dead executors on jenkins 1.622.</p>

<p>java.lang.IllegalStateException: cannot create a build with number 2322 since that (or higher) is already in use among <span class="error">&#91;2261, 2262, 2263, 2264, 2265, 2266, 2267, 2268, 2269, 2270, 2271, 2272, 2273, 2274, 2275, 2276, 2277, 2278, 2279, 2280, 2281, 2282, 2283, 2284, 2285, 2286, 2287, 2288, 2289, 2290, 2291, 2292, 2293, 2294, 2295, 2296, 2297, 2298, 2299, 2300, 2301, 2302, 2303, 2304, 2305, 2306, 2307, 2308, 2309, 2310, 2457, 2458, 2459, 2460, 2461, 2462, 2463, 2464, 2465, 2466, 2467, 2468, 2469, 2470, 2471, 2472, 2473, 2474, 2475, 2476, 2477, 2478, 2479, 2480, 2481, 2482, 2483, 2484, 2485, 2486, 2487, 2488, 2489, 2490, 2491, 2492, 2493, 2494, 2495, 2496, 2497, 2498, 2499, 2500, 2501, 2502, 2503, 2504, 2505, 2506&#93;</span>
	at jenkins.model.lazy.AbstractLazyLoadRunMap.proposeNewNumber(AbstractLazyLoadRunMap.java:361)
	at hudson.model.RunMap.put(RunMap.java:192)
	at jenkins.model.lazy.LazyBuildMixIn.newBuild(LazyBuildMixIn.java:178)
	at hudson.model.AbstractProject.newBuild(AbstractProject.java:1019)
	at hudson.model.AbstractProject.createExecutable(AbstractProject.java:1218)
	at hudson.model.AbstractProject.createExecutable(AbstractProject.java:144)
	at hudson.model.Executor$1.call(Executor.java:335)
	at hudson.model.Executor$1.call(Executor.java:317)
	at hudson.model.Queue._withLock(Queue.java:1345)
	at hudson.model.Queue.withLock(Queue.java:1210)
	at hudson.model.Executor.run(Executor.java:317)</p>
</pre>
</details>

</ul>
</details>

<p>Reopening this issue.  I'm seeing this exception and a lot of dead executors on jenkins 1.622.</p>

<p>java.lang.IllegalStateException: cannot create a build with number 2322 since that (or higher) is already in use among <span class="error">&#91;2261, 2262, 2263, 2264, 2265, 2266, 2267, 2268, 2269, 2270, 2271, 2272, 2273, 2274, 2275, 2276, 2277, 2278, 2279, 2280, 2281, 2282, 2283, 2284, 2285, 2286, 2287, 2288, 2289, 2290, 2291, 2292, 2293, 2294, 2295, 2296, 2297, 2298, 2299, 2300, 2301, 2302, 2303, 2304, 2305, 2306, 2307, 2308, 2309, 2310, 2457, 2458, 2459, 2460, 2461, 2462, 2463, 2464, 2465, 2466, 2467, 2468, 2469, 2470, 2471, 2472, 2473, 2474, 2475, 2476, 2477, 2478, 2479, 2480, 2481, 2482, 2483, 2484, 2485, 2486, 2487, 2488, 2489, 2490, 2491, 2492, 2493, 2494, 2495, 2496, 2497, 2498, 2499, 2500, 2501, 2502, 2503, 2504, 2505, 2506&#93;</span><br/>
	at jenkins.model.lazy.AbstractLazyLoadRunMap.proposeNewNumber(AbstractLazyLoadRunMap.java:361)<br/>
	at hudson.model.RunMap.put(RunMap.java:192)<br/>
	at jenkins.model.lazy.LazyBuildMixIn.newBuild(LazyBuildMixIn.java:178)<br/>
	at hudson.model.AbstractProject.newBuild(AbstractProject.java:1019)<br/>
	at hudson.model.AbstractProject.createExecutable(AbstractProject.java:1218)<br/>
	at hudson.model.AbstractProject.createExecutable(AbstractProject.java:144)<br/>
	at hudson.model.Executor$1.call(Executor.java:335)<br/>
	at hudson.model.Executor$1.call(Executor.java:317)<br/>
	at hudson.model.Queue._withLock(Queue.java:1345)<br/>
	at hudson.model.Queue.withLock(Queue.java:1210)<br/>
	at hudson.model.Executor.run(Executor.java:317)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/26739?focusedId=233760&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-233760">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-26739?focusedId=233760&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-233760">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This issue is specifically for Matrix Projects. Your problem is with a different project type. When filing a new issue (assuming none exists yet), please try to determine why Jenkins would assign a build number much lower than the highest existing build number.</p>
</pre>
</details>

</ul>
</details>

<p>This issue is specifically for Matrix Projects. Your problem is with a different project type. When filing a new issue (assuming none exists yet), please try to determine why Jenkins would assign a build number much lower than the highest existing build number.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `is_duplicated_by` from Jira: <a href="https://github.com/jenkinsci/matrix-project-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-27081%22">JENKINS-27081</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `is_duplicated_by` from Jira: <a href="https://github.com/jenkinsci/matrix-project-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-26616%22">JENKINS-26616</a>]</i>
